### PR TITLE
Add shared VSCode config.

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #1647.

## Relevant technical choices

Added a few extra settings I had locally that help with editor auto-formatting.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by ryihan — DISMISSED** [review]

**Review by eugene-manuilov — APPROVED** [review]
LGTM

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "search.exclude": {
+    ".storybook/data": true,
+    "assets/vendor": true,
```
Comment (eugene-manuilov) [inline .vscode/settings.json:4]: We don't have the `vendor` directory in the `assets` folder. Let's remove this line. Plus let's add `vendor/`, `third-party/` and `node_modules/` paths to exclude from the search.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Overall it looks good. Just found one tiny issue with the configuration.

File: .vscode/settings.json (hunk 1 above)
Comment (tofumatt) [reply .vscode/settings.json:4]: Whoops, indeed! I think that was part of an older config maybe? Part of this config is what's been on my machine a long time 😅 

Thanks, changed!

**Review by eugene-manuilov — DISMISSED** [review]
Awesome! Thanks, @tofumatt!

File: .vscode/settings.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "third-party/": true,
+    "vendor/": true
+  },
+  "files.exclude": {
+    "dist/": true,
+    "node_modules/": true,
+    "third-party/": true,
+    "vendor/": true
+  },
+  "editor.formatOnSave": true,
```
Comment (aaemnnosttv) [inline .vscode/settings.json:15]: I have this set to `false` and I still get the format on save (I believe from `editor.codeActionsOnSave`) – is it needed?

**Review by aaemnnosttv — COMMENTED** [review]
Looks good for the most part. A few questions/suggestions below @tofumatt

File: .vscode/settings.json (hunk 2 above)
Comment (tofumatt) [reply .vscode/settings.json:15]: Looks like you're right—removed! 👍🏻

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  "javascript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis": true,
+  "javascript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets": true,
+  "javascript.format.insertSpaceBeforeFunctionParenthesis": true,
+  "typescript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyBrackets": true,
+  "typescript.format.insertSpaceAfterOpeningAndBeforeClosingNonemptyParenthesis": true,
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply .vscode/settings.json:20]: Seems not, you're right. Some of this was from my own config in case it helped, but it seems it doesn't matter.

**Review by aaemnnosttv — APPROVED** [review]
LGTM 👍

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

**Review by aaemnnosttv — APPROVED** [review]
👌

**Pull request merged**